SEP-2200: Clarify tool result content visibility (#2200)
modelcontextprotocol/modelcontextprotocol: docs/specification/draft/server/tools.mdx

#### Structured Content

Tool results can include both a model-oriented representation and a machine-oriented representation:

> - `content` is the model-oriented representation of the result. Clients **SHOULD** use this field

Reviewer: The Tool Result intro a few paragraphs up still says "**Unstructured** content is returned in the `content` field," which now contradicts this section's "model-oriented" framing. The diff deleted the matching "**Structured** content is returned as JSON…" sentence but left its sibling behind.

  when providing tool results to conversational or agentic model context, if it is present.
- `structuredContent` is the machine-oriented representation of the same result for programmatic
  consumers, typed orchestration, and UI hydration.

When both `content` and `structuredContent` are present, they **SHOULD** be semantically
equivalent. `content` may summarize or textualize `structuredContent` without reproducing every
field verbatim, but it **SHOULD NOT** contradict it.

For backwards compatibility, a tool that returns structured content **SHOULD** also return a
`content` representation of the same information. JSON-serialized text is acceptable, but it is
not required and may be suboptimal depending on the use case.

> If `content` is missing or empty, clients **MAY** fall back to `structuredContent`. Clients

Reviewer: `content` is required in the schema (no `?`, `required:["content"]`), so "missing" describes only schema-invalid payloads. Same with the schema.ts JSDoc's "if it is present" and the SEP's "omitted." Only `content: []` is a valid sentinel.

```suggestion
If `content` is empty, clients **MAY** fall back to `structuredContent`. Clients
```

**SHOULD NOT** forward both fields verbatim to the model as separate inputs.

When `isError` is `true`, the error message **SHOULD** be returned in `content`. Servers **MAY**
omit `structuredContent` for error results.

#### Output Schema

Tools may also provide an output schema for validation of `structuredContent`. This schema applies
to the structured, machine-oriented representation and does not constrain the format of `content`.

> If an output schema is provided and a non-error result includes `structuredContent`:

Reviewer: I think this accidentally drops the success-path presence requirement. The current text says "If an output schema is provided: Servers MUST provide structured results that conform" (declaring `outputSchema` obligates the server to include conforming `structuredContent`). Moving "result includes `structuredContent`" into the condition means a tool can declare `outputSchema`, return success with only `content`, and the MUST never applies. The SEP only authorizes the `isError` carve-out, not relaxing presence on success.

```suggestion
If an output schema is provided and the result is not an error:
```
